homeassistant/components/sensor/dominionenergy.py

```diff
@@ -0,0 +1,79 @@
+"""
+Support for Washington Dominion Energy.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.dominion_energy/
+"""
+
+import logging
+import voluptuous as vol
+
+
+from homeassistant.util import Throttle
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_PASSWORD, CONF_USERNAME)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['selenium==3.11.0']
+
+DEFAULT_NAME = "Dominion Energy"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and getting the username and password."""
+    add_devices([DominionEnergySensor(config[CONF_NAME], config[CONF_USERNAME],
```

Comment:
You should check if the provided credentials are valid during the platform setup and fail if they are not. Otherwise the users end up with a non-working platform.